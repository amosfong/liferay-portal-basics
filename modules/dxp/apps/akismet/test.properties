##
## Test Suites
##

    modified.files.includes[relevant][akismet-functional-rule]=**

    modified.files.includes[relevant][akismet-java-rule]=\
        **/*.java,\
        **/test/**,\
        **/testIntegration/**

    relevant.rule.names=\
        akismet-functional-rule,\
        akismet-java-rule

    test.batch.names[relevant][akismet-functional-rule]=\
        functional-tomcat90-mysql57

    test.batch.names[relevant][akismet-java-rule]=\
        modules-integration-mysql57,\
        modules-unit

    #
    # Portal Hotfix Release
    #

    test.batch.run.property.query[functional-bundle-tomcat-mysql57][test-portal-hotfix-release]=\
        ${test.batch.run.property.query[functional-tomcat*-mysql*][relevant][akismet-functional-rule]}

    #
    # Relevant
    #

    modules.includes.required.test.batch.class.names.includes[modules-integration-mysql57][relevant][akismet-java-rule]=\
        dxp/apps/akismet/**/*Test.java

    modules.includes.required.test.batch.class.names.includes[modules-unit][relevant][akismet-java-rule]=\
        dxp/apps/akismet/**/*Test.java

    test.batch.run.property.query[functional-tomcat*-mysql*][relevant][akismet-functional-rule]=\
        (testray.main.component.name ~ "Akismet")

##
## Testray
##

    testray.main.component.name=Akismet