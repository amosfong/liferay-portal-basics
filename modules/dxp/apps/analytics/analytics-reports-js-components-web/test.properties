##
## Test Suites
##

    modified.files.includes[relevant][analytics-reports-js-components-web-playwright-rule]=**
    relevant.rule.names=analytics-reports-js-components-web-playwright-rule

    test.batch.names[relevant][analytics-reports-js-components-web-playwright-rule]=\
        playwright-js-tomcat90-mysql57

    #
    # Relevant
    #

    playwright.projects.includes[playwright-js-tomcat90-mysql57][relevant][analytics-reports-js-components-web-playwright-rule]=\
        analytics-reports-js-components-web