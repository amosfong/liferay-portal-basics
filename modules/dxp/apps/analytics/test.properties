##
## Test Suites
##

    modified.files.includes[relevant][analytics-functional-rule]=**

    modified.files.includes[relevant][analytics-java-rule]=\
        **/*.java,\
        **/test/**,\
        **/testIntegration/**

    relevant.rule.names=\
        analytics-functional-rule,\
        analytics-java-rule

    test.batch.names[relevant][analytics-functional-rule]=\
        functional-tomcat90-mysql57

    test.batch.names[relevant][analytics-java-rule]=\
        modules-integration-mysql57,\
        modules-unit

    #
    # Portal Hotfix Release
    #

    test.batch.run.property.query[functional-bundle-tomcat-mysql57][test-portal-hotfix-release]=\
        ${test.batch.run.property.query[functional-tomcat*-mysql*][relevant][analytics-functional-rule]}

    #
    # Relevant
    #

    modules.includes.required.test.batch.class.names.includes[modules-integration-mysql57][relevant][analytics-java-rule]=\
        apps/analytics/**/*Test.java,\
        apps/content-dashboard/**/*Test.java,\
        apps/segments/**/*Test.java,\
        dxp/apps/analytics/**/*Test.java

    modules.includes.required.test.batch.class.names.includes[modules-unit][relevant][analytics-java-rule]=\
        apps/analytics/**/*Test.java,\
        apps/content-dashboard/**/*Test.java,\
        apps/segments/**/*Test.java,\
        dxp/apps/analytics/**/*Test.java

    test.batch.run.property.query[functional-tomcat*-mysql*][relevant][analytics-functional-rule]=\
        (portal.acceptance == true) AND \
        (\
            (testray.component.names ~ "A/B Test") OR \
            (testray.component.names ~ "Personalization") OR \
            (testray.component.names ~ "Segmentation") OR \
            (testray.main.component.name ~ "A/B Test") OR \
            (testray.main.component.name ~ "Content Dashboard") OR \
            (testray.main.component.name ~ "Content Pages") OR \
            (testray.main.component.name ~ "Display Page Templates") OR \
            (testray.main.component.name ~ "Fragment Administration") OR \
            (testray.main.component.name ~ "Headless USM") OR \
            (testray.main.component.name ~ "Master Pages") OR \
            (testray.main.component.name ~ "Page Administration") OR \
            (testray.main.component.name ~ "Personalization") OR \
            (testray.main.component.name ~ "Segmentation")\
        )

##
## Testray
##

    testray.main.component.name=Analytics Cloud