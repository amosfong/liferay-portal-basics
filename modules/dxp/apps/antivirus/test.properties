##
## Test Suites
##

    modified.files.includes[relevant][antivirus-java-rule]=\
        **/*.java,\
        **/test/**,\
        **/testIntegration/**

    relevant.rule.names=antivirus-java-rule

    test.batch.names[relevant][antivirus-java-rule]=\
        modules-integration-mysql57,\
        modules-unit

    #
    # Relevant
    #

    modules.includes.required.test.batch.class.names.includes[modules-integration-mysql57][relevant][antivirus-java-rule]=\
        dxp/apps/antivirus/**/*Test.java

    modules.includes.required.test.batch.class.names.includes[modules-unit][relevant][antivirus-java-rule]=\
        dxp/apps/antivirus/**/*Test.java

##
## Testray
##

    testray.main.component.name=Antivirus