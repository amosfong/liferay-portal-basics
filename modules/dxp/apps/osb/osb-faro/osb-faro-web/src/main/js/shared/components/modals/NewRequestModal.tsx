import * as API from 'shared/api';
import ClayButton from '@clayui/button';
import ClayLink from '@clayui/link';
import ClayMultiSelect from '@clayui/multi-select';
import FileDropTarget from 'shared/components/FileDropTarget';
import Form from 'shared/components/form';
import getCN from 'classnames';
import Input from 'shared/components/Input';
import Loading, {Align} from 'shared/components/Loading';
import Modal from 'shared/components/modal';
import React, {useEffect, useRef, useState} from 'react';
import {Formik, FormikValues} from 'formik';
import {NetworkStatus} from '@clayui/data-provider';
import {paginationDefaults} from 'shared/util/pagination';
import {sub} from 'shared/util/lang';
import {useDebounce} from 'shared/hooks/useDebounce';

const SAMPLE_CSV = 'user@example.com\nuser1@example.com\nuser2@example.com';

const {page} = paginationDefaults;

const AUTOCOMPLETE_DELTA = 5;

enum SubjectIdType {
	ByEmail = 'email',
	ByFile = 'file'
}

const getCheckboxLabel = (title: string, subtitle: string): React.ReactNode => (
	<span className='checkbox-label'>
		{title}

		<span className='text-secondary'>{` - ${subtitle}`}</span>
	</span>
);

interface INewRequestModalProps {
	groupId: string;
	onClose: () => void;
	onSubmit: (value: {
		emailAddresses?: string[];
		fileName?: string;
		types: string[];
	}) => void;
}

const NewRequestModal: React.FC<INewRequestModalProps> = ({
	groupId,
	onClose,
	onSubmit
}) => {
	const [items, setItems] = useState([]);
	const [fileName, setFileName] = useState(null);
	const [email, setEmail] = useState('');
	const [networkStatus, setNetworkStatus] = useState(NetworkStatus.Unused);
	const [emails, setEmails] = useState([]);

	const debouncedEmail = useDebounce(email, 500);

	useEffect(() => {
		setNetworkStatus(NetworkStatus.Loading);

		async function fetchIndividuals() {
			const {items} = await API.individuals.search({
				delta: AUTOCOMPLETE_DELTA,
				filter: debouncedEmail
					? `contains(demographics/email/value, '${debouncedEmail}')`
					: '',
				groupId,
				page
			});

			setItems(
				items.map(({id, properties: {email}}) => ({
					label: email,
					value: id
				}))
			);
			setNetworkStatus(NetworkStatus.Unused);
		}

		fetchIndividuals();
	}, [debouncedEmail]);

	const _formRef = useRef<Formik>();

	const handleAccessClick = event => {
		const {checked} = event.target;

		if (_formRef.current) {
			const {setFieldValue} = _formRef.current;

			setFieldValue('accessRequest', checked);
		}
	};

	const handleDeleteClick = event => {
		const {checked} = event.target;

		if (_formRef.current) {
			const {setFieldValue} = _formRef.current;

			setFieldValue('deleteRequest', checked);

			setFieldValue('suppressRequest', checked);
		}
	};

	const handleFileChange = file => {
		if (file) {
			const {completed, response, status} = file;

			const fileUploaded = completed && status !== 500;

			setFileName(fileUploaded ? response : null);
		} else {
			setFileName(null);
		}
	};

	const handleSubmit = ({
		accessRequest,
		deleteRequest,
		subjectIdType,
		suppressRequest
	}) => {
		let types = [];

		if (accessRequest) {
			types = [...types, 'ACCESS'];
		}

		if (suppressRequest) {
			types = [...types, 'SUPPRESS'];
		}

		if (deleteRequest) {
			types = [...types, 'DELETE'];
		}

		if (subjectIdType === SubjectIdType.ByEmail) {
			onSubmit({
				emailAddresses: emails.map(({label}) => label),
				types
			});
		} else {
			onSubmit({
				fileName,
				types
			});
		}
	};

	const handleSuppressClick = event => {
		const {checked} = event.target;

		if (_formRef.current) {
			const {
				setFieldValue,
				state: {
					values: {deleteRequest}
				}
			} = _formRef.current;

			if (!deleteRequest) {
				setFieldValue('suppressRequest', checked);
			}
		}
	};

	const isValid = ({
		accessRequest,
		deleteRequest,
		subjectIdType,
		suppressRequest
	}: FormikValues): boolean => {
		const requests = accessRequest || deleteRequest || suppressRequest;

		const byEmail =
			subjectIdType === SubjectIdType.ByEmail && emails.length;
		const byFile = subjectIdType === SubjectIdType.ByFile && fileName;

		return requests && (byEmail || byFile);
	};

	return (
		<Modal className='new-request-modal-root'>
			<Modal.Header
				onClose={onClose}
				title={Liferay.Language.get('new-request')}
			/>

			<Form
				initialValues={{
					accessRequest: false,
					deleteRequest: false,
					subjectIdType: SubjectIdType.ByEmail,
					suppressRequest: false
				}}
				onSubmit={handleSubmit}
				ref={_formRef}
			>
				{({handleSubmit, isSubmitting, values}) => (
					<Form.Form onSubmit={handleSubmit}>
						<Modal.Body>
							<p className='text-secondary'>
								{Liferay.Language.get(
									'new-requests-will-be-added-to-the-queue-and-you-will-be-notified-once-the-job-has-completed-running'
								)}
							</p>

							<Form.Group>
								<div className='h4'>
									{Liferay.Language.get('job-type')}
								</div>

								<Form.GroupItem>
									<Form.Checkbox
										autoFocus
										label={getCheckboxLabel(
											Liferay.Language.get('access'),
											Liferay.Language.get(
												'creates-downloadable-file-of-all-the-data-collected-related-to-a-user'
											)
										)}
										name='accessRequest'
										onChange={handleAccessClick}
										value='accessRequest'
									/>
								</Form.GroupItem>

								<Form.GroupItem>
									<Form.Checkbox
										label={getCheckboxLabel(
											Liferay.Language.get('delete'),
											Liferay.Language.get(
												'delete-users-pii-and-add-them-to-the-suppression-list'
											)
										)}
										name='deleteRequest'
										onChange={handleDeleteClick}
										value='deleteRequest'
									/>
								</Form.GroupItem>

								<Form.GroupItem>
									<Form.Checkbox
										disabled={values.deleteRequest}
										label={getCheckboxLabel(
											Liferay.Language.get('suppress'),
											Liferay.Language.get(
												'suppress-identity-resolution-of-users-based-on-their-email'
											)
										)}
										name='suppressRequest'
										onChange={handleSuppressClick}
										value='suppressRequest'
									/>
								</Form.GroupItem>
							</Form.Group>

							<Form.Group>
								<div className='h4'>
									{Liferay.Language.get('data-subject-id')}
								</div>

								<Form.GroupItem>
									<Form.RadioGroup name='subjectIdType'>
										<Form.RadioGroup.Option
											key={SubjectIdType.ByEmail}
											label={Liferay.Language.get(
												'find-by-email'
											)}
											value={SubjectIdType.ByEmail}
										/>

										<Form.RadioGroup.Subsection>
											<Input.Group>
												<ClayMultiSelect
													allowsCustomLabel={false}
													disabled={
														values.subjectIdType ===
														SubjectIdType.ByFile
													}
													loadingState={networkStatus}
													onChange={setEmail}
													onItemsChange={setEmails}
													placeholder={Liferay.Language.get(
														'example-email'
													)}
													sourceItems={items}
													value={email}
												/>
											</Input.Group>
										</Form.RadioGroup.Subsection>

										<Form.RadioGroup.Option
											key={SubjectIdType.ByFile}
											label={Liferay.Language.get(
												'upload-file'
											)}
											value={SubjectIdType.ByFile}
										/>

										<Form.RadioGroup.Subsection
											className={getCN({
												hide:
													values.subjectIdType ===
													SubjectIdType.ByEmail
											})}
										>
											<div>
												<FileDropTarget
													fileTypes={['.csv']}
													onChange={handleFileChange}
													uploadURL={`/o/proxy/download/data-control-tasks?projectGroupId=${groupId}`}
												/>

												<div className='example-file-text text-secondary'>
													{sub(
														Liferay.Language.get(
															'please-upload-files-in-csv-format.-a-sample-file-can-be-found-x'
														),
														[
															<ClayLink
																download='example_user_request.csv'
																// @ts-ignore
																externalLink
																href={`data:text/octet-stream;charset=utf-8,${SAMPLE_CSV}`}
																key='EXAMPLE_FILE'
															>
																{Liferay.Language.get(
																	'here-fragment'
																)}
															</ClayLink>
														],
														false
													)}
												</div>
											</div>
										</Form.RadioGroup.Subsection>
									</Form.RadioGroup>
								</Form.GroupItem>
							</Form.Group>
						</Modal.Body>

						<Modal.Footer>
							<ClayButton
								className='button-root'
								displayType='secondary'
								onClick={onClose}
							>
								{Liferay.Language.get('cancel')}
							</ClayButton>

							<ClayButton
								className='button-root'
								disabled={!isValid(values)}
								displayType='primary'
								type='submit'
							>
								{isSubmitting && <Loading align={Align.Left} />}

								{Liferay.Language.get('save')}
							</ClayButton>
						</Modal.Footer>
					</Form.Form>
				)}
			</Form>
		</Modal>
	);
};

export default NewRequestModal;
