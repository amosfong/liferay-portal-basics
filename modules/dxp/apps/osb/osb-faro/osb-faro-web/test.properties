##
## Test Suites
##

    modified.files.includes[relevant][osb-faro-web-playwright-rule]=**
    relevant.rule.names=osb-faro-web-playwright-rule

    test.batch.names[relevant][osb-faro-web-playwright-rule]=\
        playwright-js-tomcat90-mysql57

    #
    # Relevant
    #

    playwright.projects.includes[playwright-js-tomcat90-mysql57][relevant][osb-faro-web-playwright-rule]=\
        osb-faro-web

##
## Testray
##

    testray.main.component.name=Analytics Cloud