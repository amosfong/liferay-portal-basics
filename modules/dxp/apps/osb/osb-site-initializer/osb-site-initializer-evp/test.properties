##
## Test Suites
##

    modified.files.includes[relevant][osb-site-initializer-evp-functional-rule]=**
    relevant.rule.names=osb-site-initializer-evp-functional-rule

    test.batch.names[relevant][osb-site-initializer-evp-functional-rule]=\
        functional-tomcat90-mysql57

    #
    # Portal Hotfix Release
    #

    test.batch.run.property.query[functional-bundle-tomcat-mysql57][test-portal-hotfix-release]=\
        ${test.batch.run.property.query[functional-tomcat*-mysql*][relevant][osb-site-initializer-evp-functional-rule]}

    #
    # Relevant
    #

    test.batch.run.property.query[functional-tomcat*-mysql*][relevant][osb-site-initializer-evp-functional-rule]=\
        (portal.acceptance == true) AND \
        (\
            (testray.component.names ~ "OSB Site Initializer EVP") OR \
            (testray.main.component.name ~ "OSB Site Initializer EVP")\
        )

##
## Testray
##

    testray.main.component.name=OSB Site Initializer EVP