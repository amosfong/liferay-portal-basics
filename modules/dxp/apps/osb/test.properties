##
## Test Suites
##

    modified.files.includes[relevant][osb-java-rule]=\
        **/*.java,\
        **/test/**,\
        **/testIntegration/**

    relevant.rule.names=osb-java-rule

    test.batch.names[relevant][osb-java-rule]=\
        modules-integration-mysql57,\
        modules-unit

    #
    # Relevant
    #

    modules.includes.required.test.batch.class.names.includes[modules-integration-mysql57][relevant][osb-java-rule]=\
        dxp/apps/osb/**/*Test.java

    modules.includes.required.test.batch.class.names.includes[modules-unit][relevant][osb-java-rule]=\
        dxp/apps/osb/**/*Test.java

##
## Testray
##

    testray.main.component.name=Analytics Cloud