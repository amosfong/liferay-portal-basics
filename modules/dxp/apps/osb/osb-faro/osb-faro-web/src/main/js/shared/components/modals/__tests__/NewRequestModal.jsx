import mockStore from 'test/mock-store';
import NewRequestModal from '../NewRequestModal';
import React from 'react';
import {cleanup, fireEvent, render} from '@testing-library/react';
import {DndProvider} from 'react-dnd';
import {HTML5Backend} from 'react-dnd-html5-backend';
import {Provider} from 'react-redux';
import {waitForLoadingToBeRemoved} from 'test/helpers';

jest.unmock('react-dom');

describe('NewRequestModal', () => {
	afterEach(cleanup);

	it('renders', async () => {
		const {container} = render(
			<Provider store={mockStore()}>
				<DndProvider backend={HTML5Backend}>
					<NewRequestModal />
				</DndProvider>
			</Provider>
		);

		await waitForLoadingToBeRemoved(container, {
			selector: '.loading-animation'
		});

		expect(container).toMatchSnapshot();
	});

	it('checks both Delete & Suppress checkbox when Delete is clicked', async () => {
		const {container, getByLabelText} = render(
			<Provider store={mockStore()}>
				<DndProvider backend={HTML5Backend}>
					<NewRequestModal />
				</DndProvider>
			</Provider>
		);

		await waitForLoadingToBeRemoved(container, {
			selector: '.loading-animation'
		});

		const deleteCheckbox = getByLabelText(/Delete/);
		const suppressCheckbox = getByLabelText(/Suppress/);

		expect(deleteCheckbox.checked).toBeFalse();
		expect(suppressCheckbox.checked).toBeFalse();

		fireEvent.click(deleteCheckbox);

		expect(deleteCheckbox.checked).toBeTrue();
		expect(suppressCheckbox.checked).toBeTrue();
	});

	it('checks if download sample CSV link is present', async () => {
		const {container} = render(
			<Provider store={mockStore()}>
				<DndProvider backend={HTML5Backend}>
					<NewRequestModal />
				</DndProvider>
			</Provider>
		);

		await waitForLoadingToBeRemoved(container, {
			selector: '.loading-animation'
		});

		const anchor = container.querySelector('div.example-file-text a');

		expect(anchor).toHaveAttribute(
			'href',
			'data:text/octet-stream;charset=utf-8,user@example.com\nuser1@example.com\nuser2@example.com'
		);
	});
});
