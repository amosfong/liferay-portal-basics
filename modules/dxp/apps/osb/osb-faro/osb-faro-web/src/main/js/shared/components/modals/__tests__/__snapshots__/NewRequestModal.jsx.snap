// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewRequestModal renders 1`] = `
<div>
  <div
    aria-modal="true"
    class="modal-dialog new-request-modal-root"
    role="dialog"
  >
    <div
      class="modal-content"
    >
      <div
        class="modal-header has-title"
      >
        <div
          class="h4 modal-title"
        >
          New Request
        </div>
      </div>
      <form
        class="form-root"
      >
        <div
          class="modal-body"
        >
          <p
            class="text-secondary"
          >
            New requests will be added to the queue and you will be notified once the job has completed running.
          </p>
          <div
            class="form-group"
          >
            <div
              class="h4"
            >
              Job Type
            </div>
            <div
              class="form-group-item"
            >
              <div
                class="custom-control custom-checkbox"
              >
                <label>
                  <input
                    class="custom-control-input"
                    form="[object Object]"
                    name="accessRequest"
                    type="checkbox"
                    value="false"
                  />
                  <span
                    class="custom-control-label"
                  >
                    <span
                      class="custom-control-label-text"
                    >
                      <span
                        class="checkbox-label"
                      >
                        Access
                        <span
                          class="text-secondary"
                        >
                           - Creates downloadable file of all the data collected related to a user.
                        </span>
                      </span>
                    </span>
                  </span>
                </label>
              </div>
            </div>
            <div
              class="form-group-item"
            >
              <div
                class="custom-control custom-checkbox"
              >
                <label>
                  <input
                    class="custom-control-input"
                    form="[object Object]"
                    name="deleteRequest"
                    type="checkbox"
                    value="false"
                  />
                  <span
                    class="custom-control-label"
                  >
                    <span
                      class="custom-control-label-text"
                    >
                      <span
                        class="checkbox-label"
                      >
                        Delete
                        <span
                          class="text-secondary"
                        >
                           - Delete user's PII and add them to the suppression list.
                        </span>
                      </span>
                    </span>
                  </span>
                </label>
              </div>
            </div>
            <div
              class="form-group-item"
            >
              <div
                class="custom-control custom-checkbox"
              >
                <label>
                  <input
                    class="custom-control-input"
                    form="[object Object]"
                    name="suppressRequest"
                    type="checkbox"
                    value="false"
                  />
                  <span
                    class="custom-control-label"
                  >
                    <span
                      class="custom-control-label-text"
                    >
                      <span
                        class="checkbox-label"
                      >
                        Suppress
                        <span
                          class="text-secondary"
                        >
                           - Suppress identity resolution of users based on their email.
                        </span>
                      </span>
                    </span>
                  </span>
                </label>
              </div>
            </div>
          </div>
          <div
            class="form-group"
          >
            <div
              class="h4"
            >
              Data Subject ID
            </div>
            <div
              class="form-group-item"
            >
              <div>
                <div
                  class="radio-group-root"
                >
                  <div
                    class="custom-control custom-radio"
                  >
                    <label>
                      <input
                        checked=""
                        class="custom-control-input"
                        name="subjectIdType"
                        type="radio"
                      />
                      <span
                        class="custom-control-label"
                      >
                        <span
                          class="custom-control-label-text"
                        >
                          Find by Email
                        </span>
                      </span>
                    </label>
                  </div>
                  <div
                    class="radio-group-subsection-root"
                  >
                    <div
                      class="input-group"
                    >
                      <div
                        class="form-control form-control-tag-group input-group"
                      >
                        <div
                          class="input-group-item"
                        >
                          <div
                            class="input-group"
                          >
                            <div
                              class="input-group-item d-contents input-group-item-shrink input-group-prepend"
                              role="grid"
                            />
                            <div
                              class="input-group-item input-group-prepend"
                            >
                              <input
                                aria-autocomplete="list"
                                aria-controls="clay-id-2"
                                aria-expanded="false"
                                autocomplete="off"
                                autocorrect="off"
                                class="form-control-inset"
                                placeholder="example@email.com"
                                role="combobox"
                                spellcheck="false"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                        <div
                          class="sr-only"
                        >
                          <span
                            id="clay-id-1"
                          >
                            Press backspace to delete the current row.
                          </span>
                          <span
                            aria-live="polite"
                            aria-relevant="text"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="custom-control custom-radio"
                  >
                    <label>
                      <input
                        class="custom-control-input"
                        name="subjectIdType"
                        type="radio"
                      />
                      <span
                        class="custom-control-label"
                      >
                        <span
                          class="custom-control-label-text"
                        >
                          Upload File
                        </span>
                      </span>
                    </label>
                  </div>
                  <div
                    class="radio-group-subsection-root hide"
                  >
                    <div>
                      <div
                        class="file-drop-target-root"
                      >
                        <div
                          class="drop-target-container"
                        >
                          <button
                            class="button-root btn btn-sm btn-secondary"
                            type="button"
                          >
                            Select File
                          </button>
                          <div
                            class="drop-zone"
                          >
                            Or Drag and Drop Files Here
                          </div>
                        </div>
                      </div>
                      <div
                        class="example-file-text text-secondary"
                      >
                        Please upload files in CSV format. A sample file can be found 
                        <a
                          class=""
                          download="example_user_request.csv"
                          href="data:text/octet-stream;charset=utf-8,user@example.com
user1@example.com
user2@example.com"
                        >
                          here
                        </a>
                        .
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="modal-footer"
        >
          <button
            class="button-root btn btn-secondary"
            type="button"
          >
            Cancel
          </button>
          <button
            class="button-root btn btn-primary"
            disabled=""
            type="submit"
          >
            Save
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
